frontend/src/scenes/surveys/surveyLogic.test.ts

```diff
@@ -49,7 +49,52 @@ const SURVEY: Survey = {
     targeting_flag_filters: undefined,
 }
 
-describe('survey logic', () => {
+const SINGLE_CHOICE_SURVEY: Survey = {
+    id: '118b22a3-09b1-0000-2f5b-1bd8352ceec9',
+    name: 'Single Choice survey',
+    description: '',
+    type: SurveyType.Popover,
+    linked_flag: null,
+    linked_flag_id: null,
+    targeting_flag: null,
+    questions: [
+        {
+            type: SurveyQuestionType.SingleChoice,
+            choices: ['Yes', 'No', 'OPENlabel=Maybe (explain)'],
```

Comment:
I think we might start requiring more information to be added to our choices than an array of strings can handle. This is on us so I'm happy to update that part myself 😄 unless you really want to do it which is also awesome.

What I'm proposing is to add another property to the question object that's something like:
`choicesProperties: [{ choiceId: 2, openTextOptional: true }]` where it provides more details about the choice instead of having to rely on parsing the string for the `OPENlabel` prefix.

This could also be useful when we also implement logic branching survey questions that depend on specific choices, but basically it'd be nice to have some way of adding properties to answers.

cc @neilkakkar

Reply:
makes sense! Probably easier to always index by the id than having an array with key choiceId, so:

```
choiceProperties: {
 <id: number 0 to len(choices)-1>: { attributes }
 }
 ```
 
 .. although, it's a bit janky to keep these two different things in sync, that closely depend on each other, so it's worth deprecating this in the future and move to a choices array of objects, which has the label value, and all the other attributes.
 
 For example, if I delete a choice in the middle, the indexes of all choices after that choice need to change.
 
 So, I'd say go down this route when we do have the other attributes & deprecate the current format in one go; and in the meanwhile since the open choice is always going to be last + not common to all choice options, keep it as a separate thing that we can incorporate into the deprecation in the future.

Reply:
The additional attributes can always be added later on, so if we know we want to add `choiceProperties` and deprecate `choices` later on anyway, why not just add it now, skip the additional `open_ended_choices` field for the question, and only have the smallest addition of `isOpenTextOption`attribute?

I think this could be okay, since we'll only need to keep the "other" choice in sync with `choiceProperties` for now

Reply:
Unless you meant to refactor `choices` into an array of objects, which in that case, then we can hold off on `choiceProperties` and the string prefix format should be a fine placeholder for now

Reply:
The reason I also thought adding `choiceProperties` would be fine was that not all choices actually need additional attributes onto it 😁 So we would only need to store properties for certain choices, at the expense of just keeping track the ID index value when answer choices get deleted/added, seems like an okay tradeoff to me to reduce bloat on the survey itself

Reply:
Only reason to wait is to figure out what attributes we need & whether they will change how the data should be stored.

But can be ok to deprecate now as well if you think they'll fit reasonably well into an object configuration, your call!

I think this bit wasn't clear: If we do deprecate, I'd suggest not adding `choiceProperties`, but a new `choices` array that is:

```
extendedChoices: [
{ label: 'yes'},
{ label: 'no'},
{ label: 'maybe', isOpenEnded: true}
]
```

so there's nothing to keep in sync, and posthog-js adjusts to use these new choices if they exist, otherwise the old ones.

So choose between this^ & `open_ended_choice` attribute, let's not do `choiceProperties` as an extra attribute. Make sense?

Reply:
ah, saw your follow up comments later - yes I meant array of object choices.

> seems like an okay tradeoff to me to reduce bloat on the survey itself

I'm not sure it will make that much of a difference for bloat. Not too keen because it's a lot more bug-prone.

Reply:
Since there would only ever be one open-ended choice in a question and it would always be the last choice in the list of choices, another option could be to keep choices as a string and add an extra field on the question itself, such as `has_open_choice`, that would force the last choice in the choices to be an open-ended choice. That way, you won't need to worry as much about introducing lots of bugs with this feature.

Reply:
Actually, since there `choices` field is just a JSON string in the database, it can just be converted to an array of objects instead of an array of strings by the API backend when retrieving, creating, or updating a survey.

When creating or updating a survey:
```python
choices = raw_question.get("choices")
if choices and not isinstance(choices, list):
  raise serializers.ValidationError("Question choices must be a list of strings")
# if each choice is an instance of a string, convert it to an object before saving to database
```

When retrieving surveys:
```python
surveys = SurveyAPISerializer(
        Survey.objects.filter(team_id=team.id).exclude(archived=True).select_related("linked_flag", "targeting_flag"),
        many=True,
).data
# convert each survey's choices to an array of objects instead of an array of strings
```

If you set up the backend to migrate the legacy survey data like this, the frontend will not have to have logic to handle both cases where choices can be an array of objects or an array of strings. The frontend will always get an array of objects. So you can isolate the location where you may introduce potential bugs to the backend safely while being able to add more potential properties to a choice in the future (or even having multiple open-ended choices in a question).

Reply:
the problem is that posthog-js consumes this as well, and every API client that creates their own surveys, so changing structure will be a breaking change. It isn't limited to our backend-frontend communication, but agree that when we do go this route, the backend can do the backwards-compatible conversion

Reply:
Ah I see, so because the consumers of the API isn't just limited to the PostHog frontend, the API must keep the original data structure of the choices field to whatever was set by the original consumer. So the logic for converting choices to an array of objects would have to happen on the consumer side. Should it convert the choices to be an array of objects for any survey updated in the frontend even if it was originally as an array of strings? Or should only new surveys have save choices as an array of objects?

Reply:
@neilkakkar I think the breaking change probably makes sense to do now, along with this new feature so it's easier in one go. I don't think there'll be too many more, if any, changes to add to the MCQ choices array object after this open text "other" field. And it'll also be nice to make this change earlier on rather than later when even more people are using it.

Reply:
Easiest to just implement the `has_open_choice` field right now then and then we'll worry about choices array deprecation as it comes later on 😅